### PR TITLE
interfere: remove unwanted replaces for VCS packages

### PR DESCRIPTION
**Pull request by ishitatsuyuki**
Ref: https://github.com/chaotic-aur/packages/issues/16

The logic is conservative so that it only applies to `replaces=(a)` for `a-git`.

Some examples of existing packages with this problem are: `booster-git`, `epiphany-git`.

**Untested.**

### CODE REVIEW

**Review by IslandC0der — APPROVED** [review]
Thank you, I think it'd be nice to have this :)

**Comment by IslandC0der** [comment]
Would you mind signing your commit and rebasing? I can't merge it otherwise :/

**Review by PedroHLC — APPROVED** [review]
Thanks for finding these bad gems. :policeman: 

@dr460nf1r3 can you rebuild them after this gets merged?
@IslandC0der I'll break this requirement here.

File: src/lib/interfere.sh [hunk 1]
```diff
@@ -78,6 +83,12 @@ function interference-generic() {
     sed -i'' 's/^groups=.*$//g' PKGBUILD
   fi
 
+  # * replaces=() (for VCS packages) generally causes unnecessary problems and should be avoided.
+  # * https://wiki.archlinux.org/title/VCS_package_guidelines#Guidelines
+  if ([ -z "${_PKGTAG_NON_VCS}" ] && grep -qP "^replaces=(${_PKGTAG_NON_VCS})$" PKGBUILD); then
```
Comment (PedroHLC) [inline src/lib/interfere.sh:88]: everything was right, except this condition :microscope:

**Pull request merged**